lab_2_retrieval_w_bm25/main.py

```diff
@@ -19,6 +21,22 @@ def tokenize(text: str) -> list[str] | None:
     In case of corrupt input arguments, None is returned.
     """
 
+    if not isinstance(text, str):
+        return None
+    tokens = []
+    for token in text.lower().split():
```

Comment:
here two loops, it is too much. Try to come up with a new way to tokenize.

Reply:
Done

Reply:
where?

Reply:
Done..?